src/ol/ol.js

```diff
@@ -160,6 +174,40 @@ ol.OVERVIEWMAP_MIN_RATIO = 0.1;
 
 
 /**
+ * @define {number} Maximum number of source tiles for raster reprojection of
+ *     a single tile.
+ *     If too many source tiles are determined to be loaded to create a single
+ *     reprojected tile the browser can become unresponsive or even crash.
+ *     This can happen if the developer defines projections improperly and/or
+ *     with unlimited extents.
+ *     If too many tiles are required, no tiles are loaded and
+ *     `ol.TileState.ERROR` state is set. Default is `100`.
+ */
+ol.RASTER_REPROJECTION_MAX_SOURCE_TILES = 100;
```

Comment:
The resolution of the source data is determined in the center of the reprojected tile and is constant for the whole tile.
If the projection behaves significantly different in the center and at the edges of the tile, the number of required tiles can be very high.

Another case (probably easier to achieve):
This can also happen when the tile grid has more than 100 tiles on the least-detailed zoom level and the reprojection needs to render the whole extent. (It would have to download and process all the tiles, potentially crashing the browser.)